activiti-cloud-service-common/activiti-cloud-services-common-security/src/main/java/org/activiti/cloud/services/common/security/jwt/JwtUserInfoUriAuthenticationConverter.java

```diff
@@ -63,10 +63,16 @@ public String getPrincipalClaimName(Jwt jwt) {
             Instant expiresAt = jwt.getExpiresAt();
             OAuth2AccessToken accessToken = new OAuth2AccessToken(TokenType.BEARER, jwt.getTokenValue(), issuedAt, expiresAt);
             OAuth2UserRequest userRequest = new OAuth2UserRequest(clientRegistration, accessToken);
-            OAuth2User oAuth2User = this.oAuth2UserServiceCacheable.loadUser(userRequest, jwt.getClaimAsString(SESSION_ID));
+            OAuth2User oAuth2User = this.oAuth2UserServiceCacheable.loadUser(userRequest, getCacheKey(jwt));
             username = oAuth2User.getName();
         }
         return username;
     }
 
+    private String getCacheKey(Jwt jwt) {
+        return jwt.hasClaim(SESSION_ID)
+            ? jwt.getClaimAsString(SESSION_ID)
+            : jwt.getTokenValue();
```

Comment:
the else statement, should it be SUB?

Reply:
ok, let's use a more fine-grained claim field for cache